dvc/utils/compat.py

```diff
@@ -27,6 +27,11 @@ def encode(u, encoding=None):
     return u.encode(encoding, "replace")
 
 
+def decode(s, encoding=None):
+    encoding = encoding or DEFAULT_ENCODING
+    return s.decode(encoding, "replace")
```

Comment:
why are we using the `replace` policy here? it looks dangerous for such a generic method that will be used everywhere

Reply:
Because it is like so in https://github.com/ipython/ipython_genutils/blob/257a4f044e452b2e7534f67028dae9b09782e4fd/ipython_genutils/py3compat.py#L15 , where we took this from. Not sure why this is dangerous though.

Reply:
it's a destructive operation (it replace symbols, right?) For example it can make from an invalid file path another one which is valid but a different one. It's fine for use for text processing sometimes. It can be really dangerous in other situations. The best strategy for a general purpose wrapper is to fail-fast.